samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -270,6 +303,7 @@ public void onBarrierStateChanged(final String version, ZkBarrierForVersionUpgra
           // In our consensus model, if the Barrier is timed-out, then it means that one or more initial
           // participants failed to join. That means, they should have de-registered from "processors" list
           // and that would have triggered onProcessorChange action -> a new round of consensus.
+          metrics.singleBarrierRebalancingTime.update(System.nanoTime() - startTime);
```

Comment:
You can just increment this timer once before the "if" condition, instead of updating it in both if and else case :) Let me know if I am missing something obvious here.